basic_out_stream.go

```diff
@@ -92,13 +92,13 @@ func (bs *BasicOutStream) encodeAndFlush(n interface{}) error {
 	err := bs.enc.Encode(n)
 	if err != nil {
 		glog.Errorf("send message encode error for peer %v: %v", peer.IDHexEncode(bs.Stream.Conn().RemotePeer()), err)
-		return ErrOutStream
+		return err
```

Comment:
Curious, why this change? In particular, is there a reason we were returning `ErrOutStream` specifically before?

Reply:
I thought `ErrOutStream` was hiding the real error

Reply:
OK if we don't have any other behavior that assumes `ErrOutStream` here, similar to how we have special expectations for `ErrHandleMsg`